third_party/musl/src/starboard/dirent.c

```diff
@@ -0,0 +1,53 @@
+#if SB_API_VERSION < 16
+
+#include <stdio.h>
+#include <dirent.h>
+
+#include <fcntl.h>
+#include <time.h>
+#include "starboard/common/log.h"
+#include "starboard/configuration_constants.h"
+#include "starboard/directory.h"
+#include "starboard/file.h"
+
+int strlcpy_dirent(char* dst, const char* src, int dst_size) {
+  for (int i = 0; i < dst_size; ++i) {
+    if ((dst[i] = src[i]) == 0)  // We hit and copied the terminating NULL.
+      return i;
+  }
+
+  // We were left off at dst_size.  We over copied 1 byte.  Null terminate.
+  if (dst_size != 0)
+    dst[dst_size - 1] = 0;
+
+  // Count the rest of the |src|, and return its length in characters.
+  while (src[dst_size])
+    ++dst_size;
+  return dst_size;
+}
+
+DIR* opendir(const char *path)
+{
+    SbDirectory result = SbDirectoryOpen(path, NULL);
+    if (result == kSbDirectoryInvalid){
+        return NULL;
+    }
+    return &result;
+}
+
+int closedir(DIR* dir){
+    SbDirectory sbdir = (SbDirectory)dir;
```

Comment:
I don't think we can cast DIR to SbDirectory.
We may need to define a custom DIR type for SB < 16.

Reply:
The implementation where I cast the pointer wasn't working, so I defined a custom DIR type.